CLAUDE.md

```diff
@@ -117,6 +117,10 @@ Invariants are constraints that must hold across the entire codebase. Reference
 | **INV-6** | Transactions in Services | Services manage transaction boundaries, not handlers |
 | **INV-7** | Events + Projections | Events are source of truth; projections for queries; both updated in same transaction |
 | **INV-8** | Workspace Scoping | Resources belong to workspaces; workspace is the sharding boundary |
+| **INV-9** | No Singletons | Pass dependencies explicitly; no module-level state or `getInstance()` patterns |
```

Comment:
Actually there's one exception we should mention which is the logger. Otherwise this is good!